openedx/features/wikimedia_features/wikimedia_general/tasks.py

```diff
@@ -60,6 +68,73 @@ def send_thread_creation_email_task(context, is_thread, post_id):
 
     if users_list:
         receipients = [user.email for user in users_list]
-        send_thread_creation_email(receipients, context, is_thread)
+        send_thread_creation_email(receipients, contexts, is_thread)
     else:
         log.info("No user is tagged on thread/comment of discussion forum.")
+
+
+
+@task(base=LoggedTask)
+def send_weekly_digest_new_post_notification_to_instructors(threads):
+    """
+    Asynchronously sends email notifications to course instructors about new discussion posts created.
+    This function is designed to process a list of discussion posts (threads), verify if notifications
+    are configured for the site and the post type, and queue emails for eligible posts.
+
+    Args:
+        threads (list): A list of thread objects, each representing a discussion post. These threads are
+                        expected to have properties like `type`, `course_id`, and methods like `to_dict()`.
+
+    Returns:
+        None
+
+    Processes each thread in the given list, checking if it is of type 'thread' and if notifications are
+    enabled for the post's associated site and course. Builds context for each valid thread and queues 
+    a batch email sending task if there are any threads to notify.
+
+    Each thread's context includes the course key, the site, and whether it's a thread. The function logs
+    information about each processed thread and about the queuing of notifications.
+    """
+    current_site = get_current_site()
+    if current_site is None:
+        current_site = Site.objects.get_current()
+    
+    contexts = []
+    message_contexts = []
+
+    for post in threads:
+        if post.type != 'thread':
+            continue
+
+        if not is_discussion_notification_configured_for_site(current_site, post.id):
+            continue
+
+        post_id = post.course_id
+        course_key = CourseKey.from_string(post.course_id)
+        data = post.to_dict()
+        
+        context = {
+            'course_id': course_key,
+            'site': current_site,
+            'is_thread': True,
+        }
```

Comment:
Isn't this context same for every thread in the provided list in s single function call? If yes, we should move it outside of the loop and convert it from a `list` of `dicts` to just a `dict`.